src/RemoteRequest/CurlRemoteGetRequest.php

```diff
@@ -98,6 +99,13 @@ public function __construct($sslVerify = true, $timeout = self::DEFAULT_TIMEOUT,
      */
     public function get($url, $headers = [])
     {
+        try {
+            $cachedRequest  = new TemporaryFileCachedRemoteGetRequest();
+            return $cachedRequest->get($url, $headers);
```

Comment:
This should not be instantiated within the `CurlRemoteGetRequest`. Instead, where we are now instantiating the `CurlRemoteGetRequest` as the default implementation when none was provided, we should instantiate `new TemporaryFileCachedRemoteGetRequest(new CurlRemoteGetRequest())` instead. This works if, as commented elsewhere, the `TemporaryFileCachedRemoteGetRequest` is a decorator.

The cURL implementation should not know anything about the caching implementation.